### DISCUSSION TITLE
Silent UART connection error

### DISCUSSION THREAD

**Discussion by GGunnison** [nasa/fprime#784] [Q&A] [answered]
Hi everyone, I'm trying to set up a UART connection to a device and I'm getting a silent connection error while trying to open the device. The follow code is hanging while trying to connect to the ADCS uart device and thus the app is not connecting to gds. I'm on mac os 10.14.6, is this linux serial driver not compatible with my operating system? The device we are trying to interface with requires 1 stop bit. It doesn't look to me like the LinuxSerialDriverComponent supports stop bits, could this cause a silent hanging error? 

The device location on my computer is: "/dev/tty.usbserial-AQ00R2IC"

`   
if (!gpsUartDrv.open(gps_device,
		Drv::LinuxSerialDriverComponentImpl::BAUD_9600,
                Drv::LinuxSerialDriverComponentImpl::NO_FLOW,
                Drv::LinuxSerialDriverComponentImpl::PARITY_NONE,
                true))
    {
	Fw::Logger::logMsg("[ERROR] Failed to open GPS UART: %s\n", reinterpret_cast<POINTER_CAST>(gps_device));
    } else {
	Fw::Logger::logMsg("[INFO] Opened GPS UART: %s\n", reinterpret_cast<POINTER_CAST>(gps_device));
        gps_uart_connected = true;
    }
 if (!adcsUartDrv.open(adcs_device,
		Drv::LinuxSerialDriverComponentImpl::BAUD_115K,
                Drv::LinuxSerialDriverComponentImpl::NO_FLOW,
                Drv::LinuxSerialDriverComponentImpl::PARITY_NONE,
                true))
    {
	Fw::Logger::logMsg("[ERROR] Failed to open ADCS UART: %s\n", reinterpret_cast<POINTER_CAST>(adcs_device));
    } else {
	Fw::Logger::logMsg("[INFO] Opened ADCS UART: %s\n", reinterpret_cast<POINTER_CAST>(adcs_device));
        adcs_uart_connected = true;
    }`

**Comment by Joshua-Anderson** [comment]
By silent connection error do you mean that you're not receiving anything from the device or is anything else going on?

**Comment by Joshua-Anderson** [comment]
1 stop bits is the expectation, that isn't something you should need to configure. Are you If you set your baud rate, flow control, and parity to something supported by your hardware I would expect the the driver to work.

It's also possible that you're hitting some Mac OS specific quirk... while this driver should work on Mac OS, it's been primarily used on linux. I'm not sure how extensively it's been tested on Mac OS recently.

**Comment by LeStarch** [comment]
I don't know why the open call would hang, unless UART device permissions weren't set correctly.  In general, ensure you have started the receive thread and supplied buffers to the UART device.

**Comment by GGunnison** [comment, marked as answer]
Looks like the same code works on a linux based device, we've punted figuring that out for now, but there is some mac OS issue here - could be computer settings specific, but elected not to dig in further at the expense of time.

**Reply by LeStarch** [reply]
Marked this as the answer.  If you ever get to the bottom of the Mac OSX issue, please post a short follow-up.
